managed/resource_cluster_db_audit_export_config.go

```diff
@@ -0,0 +1,394 @@
+	}
+
+	return result, nil
+}
+
+func getDbAuditExporterConfigSpec(plan DbAuditExporterConfig) (*openapiclient.DbAuditExporterConfigSpec, error) {
+	statementClasses, err := convertToDbAuditYsqlStatmentClassesEnum(plan.YsqlConfig.StatementClasses)
+	if err != nil {
+		return nil, fmt.Errorf("error in converting statement classes string to DbAuditYsqlStatmentClassesEnum for cluster %s: %s", plan.ClusterID.Value, err)
+	}
+
+	dbAuditLogSettings, err := GetDbAuditYsqlLogSettings(plan)
+	if err != nil {
+		return nil, fmt.Errorf("error in obtaining LogSettings for cluster %s: %s", plan.ClusterID.Value, err)
+	}
+
+	dbAuditYsqlExportConfig := openapiclient.NewDbAuditYsqlExportConfig(statementClasses)
+	if dbAuditLogSettings != nil {
+		dbAuditYsqlExportConfig.SetLogSettings(*dbAuditLogSettings)
+	}
+
+	dbAuditExporterConfigSpec := openapiclient.NewDbAuditExporterConfigSpec(*dbAuditYsqlExportConfig, plan.ExporterID.Value)
+
+	return dbAuditExporterConfigSpec, nil
+}
+
+// Create a new Db Audit Log Configuration for a Cluster
+func (r resourceClusterDbAuditExportConfig) Create(ctx context.Context, req tfsdk.CreateResourceRequest, resp *tfsdk.CreateResourceResponse) {
+	if !r.p.configured {
+		resp.Diagnostics.AddError(
+			"Provider not configured",
+			"The provider wasn't configured before being applied, likely because it depends on an unknown value from another resource.",
+		)
+		return
+	}
+
+	var plan DbAuditExporterConfig
+	var accountId, message string
+	var getAccountOK bool
+
+	resp.Diagnostics.Append(getClusterDbAuditLogConfigPlan(ctx, req.Plan, &plan)...)
+	if resp.Diagnostics.HasError() {
+		tflog.Debug(ctx, "Error while getting the plan for the DB audit log configuration")
+		return
+	}
+
+	apiClient := r.p.client
+	accountId, getAccountOK, message = getAccountId(ctx, apiClient)
+	if !getAccountOK {
+		resp.Diagnostics.AddError("Unable to get account ID", message)
+		return
+	}
+
+	projectId, getProjectOK, message := getProjectId(ctx, apiClient, accountId)
+	if !getProjectOK {
+		resp.Diagnostics.AddError("Unable to get project ID", message)
+		return
+	}
+
+	exporterId := plan.ExporterID.Value
+
+	_, err := GetConfigByNameorID(accountId, projectId, exporterId, "", apiClient)
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to associate telemetry exporter to cluster", GetApiErrorDetails(err))
+		return
+	}
+
+	clusterId := plan.ClusterID.Value
+	_, err = GetClusterByNameorID(accountId, projectId, clusterId, "", apiClient)
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to associate metrics exporter to cluster", GetApiErrorDetails(err))
+		return
+	}
+
+	dbAuditExporterConfigSpec, err := getDbAuditExporterConfigSpec(plan)
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to obtain DbAuditExporterConfigSpec", GetApiErrorDetails(err))
+		return
+	}
+
+	response, _, err := apiClient.ClusterApi.AssociateDbAuditExporterConfig(ctx, accountId, projectId, clusterId).DbAuditExporterConfigSpec(*dbAuditExporterConfigSpec).Execute()
```

Comment:
Since this API is a task, you need to read status and wait for task to be done. For eg
```
	// read status, wait for status to be done
	readClusterRetries := 0
	retryPolicy := retry.NewConstant(10 * time.Second)
	retryPolicy = retry.WithMaxDuration(1200*time.Second, retryPolicy)
	err = retry.Do(ctx, retryPolicy, func(ctx context.Context) error {
		clusterState, readInfoOK, message := getClusterState(ctx, accountId, projectId, clusterId, apiClient)
		if readInfoOK {
			if strings.EqualFold(clusterState, "Active") {
				return nil
			}
		} else {
			return handleReadFailureWithRetries(ctx, &readClusterRetries, 2, message)
		}
		return retry.RetryableError(errors.New("Cluster CMK is getting updated."))
	})

	if err != nil {
		return errors.New("Unable to edit cluster CMK. " + "The operation timed out waiting to edit CMK.")
	}

	return nil
	```

Reply:
The retry mechanism is written [here](https://github.com/yugabyte/terraform-provider-ybm/pull/97/files#diff-ab0457bfb62f250bffb542e50d70fcf9b061938edfe3862b1e03699bf8644885R294) -